CDP4WebServices.API/Services/BusinessLogic/Implementation/OldParameterContextProvider.cs

```diff
@@ -230,5 +232,43 @@ public ParameterValueSet GetsourceValueSet(Guid? option, Guid? state)
                     ?? this.OldValueSet.FirstOrDefault(x => this.OldDefaultState != null && x.ActualState == this.OldDefaultState.Iid)
                     ?? this.OldValueSet.FirstOrDefault();
         }
+
+        /// <summary>
+        /// Gets the default value 
+        /// </summary>
+        /// <returns>The default <see cref="ParameterValueSet"/></returns>
+        public ParameterValueSet GetDefaultValue()
+        {
+            if (this.OldParameter == null)
+            {
+                return null;
+            }
+
+            if (!this.IsOldStateDependent && !this.IsOldOptionDependent)
+            {
+                return this.OldValueSet.FirstOrDefault();
+            }
+
+            // old is both state/option dependent
+            if (this.IsOldStateDependent && this.IsOldOptionDependent)
+            {
+                return this.OldDefaultState != null & this.OldDefaultOption != null
+                    ? this.OldValueSet.FirstOrDefault(x => x.ActualState == this.OldDefaultState.Iid && x.ActualOption == this.OldDefaultOption.Iid)
+                    : this.OldDefaultOption != null 
+                        ? this.OldValueSet.FirstOrDefault(x => x.ActualOption == this.OldDefaultOption.Iid)
+                        : this.OldDefaultState != null
+                            ? this.OldValueSet.FirstOrDefault(x => x.ActualState == this.OldDefaultState.Iid)
+                            : this.OldValueSet.FirstOrDefault();
```

Comment:
Of course it is about personal preference, but I do not think it is a good idea to have so many ternary operators in a return statement. Basically it is considered to be a code smell and it is hard to debug and read. As a small proof https://rules.sonarsource.com/csharp/RSPEC-3358?search=ternar